client/components/masonry-grid/index.tsx

```diff
@@ -0,0 +1,86 @@
+import clsx from 'clsx';
+import { debounce } from 'lodash';
+import { FC, PropsWithChildren, useRef, useLayoutEffect } from 'react';
+
+import './style.scss';
+
+const calculateMasonryGrid = ( wrapper: HTMLElement ) => {
```

Comment:
Not passing the `itemSelector` argument does simplify things somewhat, but it will break once we integrate this in the Pattern Library (because we have a pagination button in `PatternGalleryClient` that we don't want to touch).

Maybe we actually want to preserve that functionality after all? Flexibility seems like a good thing, especially since we're making the component reusable

Reply:
Closing the loop: we reviewed this in a sync call and decided to stick with the current approach 👍

A few reasons:
- `itemSelector` can be confusing because it implies that we can select items at a deeper level in the DOM. That's not true, though. We can only affect the layout of immediate children.
- This layout method relies heavily on CSS grid, which only lays out the immediate children of the wrapper. Sticking close to that "native approach" makes sense.